blockly-dom.js

```diff
@@ -618,26 +627,29 @@ Blockly.JavaScript["set_content"] = function (block) {
 
 Blockly.JavaScript["get_input_value_with_id"] = function (block) {
   let elementId = block.getFieldValue("ID");
-  let getNumberOrString = Blockly.JavaScript.provideFunction_(
-    "getNumberOrString",
-    [
-      "function " + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + "(value) {",
-      "  // Convert a string value to a number if possible",
-      "  let number_value = Number(value);",
-      "  if (Number.isNaN(number_value)) {",
-      "    return value",
-      "  } else {",
-      "    return number_value",
-      "  }",
-      "}",
-    ]
-  );
-  return [
-    `getNumberOrString(document.getElementById(${Blockly.JavaScript.quote_(
-      elementId
-    )}).value)`,
-    Blockly.JavaScript.ORDER_MEMBER,
-  ];
+  let inputType = block.getFieldValue("TYPE");
+
+  Blockly.JavaScript.provideFunction_("convertToNumber", [
+    "function " + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + "(value) {",
+    "  // Convert a string value to a number if possible",
+    "  let number_value = Number(value);",
+    "  if (Number.isNaN(number_value)) {",
+    "    return 0",
+    "  } else {",
+    "    return number_value",
+    "  }",
+    "}",
+  ]);
+  const element = `document.getElementById(${Blockly.JavaScript.quote_(
+    elementId
+  )})`;
+  if (inputType === "number") {
+    return [
+      `convertToNumber(${element}.value)`,
+      Blockly.JavaScript.ORDER_MEMBER,
```

Comment:
Technically, this is ORDER_FUNCTION_CALL, which is almost the same as ORDER_MEMBER (to the point I can't think of a case where it could matter)